### PR TITLE
Add support for signed arguments in range() #43333

### LINKED ISSUE #43333: Add support for signed arguments in range()
**Describe the issue**
Allow to call `range()` with Int* data types.
Validate that arguments are positive.

**How to reproduce**
```
SELECT range(CAST('1', 'Int8'), CAST('10', 'Int8'))

Query id: 8710056b-08d9-454d-bb6a-e858b61dc9a3


0 rows in set. Elapsed: 0.081 sec.

Received exception from server (version 22.11.1):
Code: 43. DB::Exception: Received from localhost:9000. DB::Exception: Illegal type Int8 of argument of function range: In scope SELECT range(CAST('1', 'Int8'), CAST('10', 'Int8')). (ILLEGAL_TYPE_OF_ARGUMENT)
```

### PR DESCRIPTION
**Pull request by wineternity**
### Changelog category (leave one):
- Improvement

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Add support for signed arguments in range(). Fixes #43333 

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

### CODE REVIEW

**Comment by wineternity** [comment]
May I ask how to check the style failure?  the script utils/check-style/check-style can not be used in Mac OS。 The format in .clang-format will be format a lot which seems strange by this line `AlignAfterOpenBracket: AlwaysBreak`

**Comment by qoega** [comment]
If you have docker installed. Easiest way for me
```
docker run -ti -v ~/clickhouse:/ClickHouse -v ~/tmp/test_output:/test_output clickhouse/style-test bash
git config --global --add safe.directory '*'
/run.sh
```

**Comment by CheSema** [comment]
As I see there is some related problems has been found by fuzzer. 
Is it clear how to cover this issues? Have you difficulties with logs?

**Comment by wineternity** [comment]
@CheSema hello ?  could we ignore the stress test failure

**Comment by CheSema** [comment]
test_distributed_queries_stress  -- https://github.com/ClickHouse/ClickHouse/issues/41776
Stress test has fatal messages -- https://github.com/ClickHouse/ClickHouse/issues/44212

File: tests/queries/0_stateless/01408_range_overflow.sql [hunk 1]
```diff
@@ -1,12 +1,13 @@
 -- executeGeneric()
-SELECT range(1025, 1048576 + 9223372036854775807, 9223372036854775807); -- { serverError 69; }
```
Comment (CheSema) [inline tests/queries/0_stateless/01408_range_overflow.sql:2]: Nice catch! 
That test was told us that there is a bug in the old realisation.

1025 and 9223372036854776832 are UInt64.
The next value is 18446744073709552639 and it is out of range UInt64. But that value do not present in result and shouldn't affect the result.

Now the behaviour is better.

Comment (wineternity) [reply tests/queries/0_stateless/01408_range_overflow.sql:2]: > 

@CheSema thank you

**Pull request merged**